test/browser/connection.test.js

```diff
@@ -218,13 +218,16 @@ define(['shared_helper', 'chai'], function (Helper, chai) {
         /* connectionKey has the form <instanceID>!<hmac>-<handleID> */
 
         /* remove the handleID from the end of key */
-        let k = key.split('-')[0];
+        const lastIndex = key.lastIndexOf('-');
+        if (lastIndex !== -1) {
+          key = key.slice(0, lastIndex);
```

Comment:
I think `key` is never declared now?
should add `let key;` before if statement

Reply:
line 217, it is declares as a parameter of function `connectionHmac`

Reply:
right, it was outside of the visible diff, didn't see it

Reply:
i do think we should revert back to `let k ... return k` - it's bad practice to mutate arguments like this

Reply:
> i do think we should revert back to let k ... return k - it's bad practice to mutate arguments like this

agree